docgen.json

```diff
@@ -1,5 +1,16 @@
 {
-  "$schema": "dist/schema.json",
-  "parseCompilerOptions": "./tsconfig.json",
-  "examplesCompilerOptions": "./tsconfig.examples.json"
+  "$schema": "./schema.json",
+  "exclude": ["bin.ts"],
+  "parseCompilerOptions": {
```

Comment:
I think we can remove this and use the default (i.e. `defaultCompilerOptions`), IMO `"exactOptionalPropertyTypes": true` is not important at all here

Reply:
If I do that then the generated docs for the config (which uses `/schema` will include `xxx | undefined` everywhere.

Reply:
So imho this is required. Unless you want the incorrect types for the config schema in the docs :-P

Reply:
opss forgot about schema... you are right, what about adding `exactOptionalPropertyTypes: true` to `defaultCompilerOptions` then?

Reply:
Then we screw up `effect` or force it to override it, would be fine by me. But `effect` doesn't work with `exactOptionalPropertyTypes`

Reply:
> But effect doesn't work with exactOptionalPropertyTypes

how's that? mmmh this is worth a discussion IMO, but not now, let's go on, so it doesn't block the other PR

Reply:
If we enable `exactOptionalPropertyTypes` in `effect`'s tsconfig.json then there's a bunch of type errors atm.

Yes, we can fix that likely. But that's sth. for Mike to weigh in on whether we want to do that or not. Imho it would be a good idea to do and be explicit about it.